File: app/src/main/java/com/worldsnas/starplayer/viewModel/MusicListViewModel.java
package com.worldsnas.starplayer.viewModel;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;

import com.worldsnas.starplayer.service.model.Music;
import com.worldsnas.starplayer.service.repository.ProjectRepository;

import java.util.List;

public class MusicListViewModel extends AndroidViewModel {

    private final LiveData<List<Music>> musicListObservable;

    public MusicListViewModel(@NonNull Application application) {
        super(application);

        musicListObservable = ProjectRepository.getInstance().getMusicList();
    }

    public LiveData<List<Music>> getMusicListObservable() {
        return musicListObservable;
    }
}

File: app/src/main/java/com/worldsnas/starplayer/viewModel/MusicListViewModel.kt
package com.worldsnas.starplayer.viewModel

import androidx.lifecycle.*
import com.worldsnas.starplayer.service.model.Music
import com.worldsnas.starplayer.service.repository.ProjectRepository
import kotlinx.coroutines.launch

class MusicListViewModel(private val repository: ProjectRepository) : ViewModel() {

    private val _musicList = MutableLiveData<List<Music>>()
    val musicList: LiveData<List<Music>> = _musicList

    fun fetchMusicList() {
        viewModelScope.launch {
            _musicList.value = repository.fetchMusicList()
        }
    }
}